gossip/gossip.go

```diff
@@ -656,17 +656,18 @@ func (g *Gossip) maybeWarnAboutInit(stopper *stop.Stopper) {
 		// This will never error because of infinite retries.
 		for r := retry.Start(retryOptions); r.Next(); {
 			g.mu.Lock()
+			hasConnections := g.outgoing.len() > 0 || g.incoming.len() > 0
 			hasSentinel := g.is.getInfo(KeySentinel) != nil
 			triedAll := g.triedAll
 			g.mu.Unlock()
 			// If we have the sentinel, exit the retry loop.
 			if hasSentinel {
 				break
-			}
-			// Otherwise, if all bootstrap hosts are connected, warn.
-			if triedAll {
-				log.Warningf("connected to gossip but missing sentinel. Has the cluster been initialized? " +
-					"Use \"cockroach init\" to initialize.")
+			} else if !hasConnections {
+				log.Warningf("not connected to gossip; check that gossip flag is set appropriately")
+			} else if triedAll {
+				// Otherwise, if there are connections, but no sentinel but all bootstrap hosts are connected, warn.
```

Comment:
"but all bootstrap hosts are connected" - does this actually check for that? this looks like an earlier version of the comment snuck in.